### PR TITLE
A4A-Partner Directory: Update the Agency Details page

### PR DESCRIPTION
**Pull request by cleacos**
Resolves: https://github.com/Automattic/automattic-for-agencies-dev/issues/663

## Proposed Changes

image

image

This PR fixes these issues:

- The Agency Details header, in the description `Want to update your expertise instead?` is a link that will take you to the Expertise form.
- The Company location is a country selector instead of a textbox.
- The Minimum project budget list is a selector instead of a textbox.
- The Budget details Optionally set your minimum budget. Clients can filter these details to find the right agency should be under the title.
- The Availability checkbox should be a switcher.
- The CTA buttons, like the `Publish public profile` should be aligned on the right.
- The dropdown selector should be larger.
- The `Publish public profile` is only active if you fill out all the fields and select at least one item on each selector.

The form selector hook will be used to map the API data and the form. There, the selector values will be mapped by their slugs.

This PR is not connected with the endpoint, so the form won't do anything. It will be addressed in a follow-up PR together with this: https://github.com/Automattic/wp-calypso/pull/91683

Any style/design issue will be addressed after the designer's review (behind feature/section) flag.

There are known issues to be fixed. The services, products, and languages selectors are working with value instead of slug. 

## Why are these changes being made?

*

## Testing Instructions

- Check the code
- Comparing with trunk:
   - Now, the Minimum project budget is a selector instead of a textbox.
   - The `Optionally set your minimum budget. Clients can filter these details to find the right agency.` is below the `Budget details` title.
   - The Availability is a switcher instead of a checkbox. Bonus: If you change the status, the text will change: `I'm accepting new clients` <=> `I'm not accepting new clients`.
   - The dropdown selectors are a bit larger (height).

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/a8c-for-agencies/sections/partner-directory/agency-details/hooks/use-country-list.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import { useTranslate } from 'i18n-calypso';
```
Comment (andrii-lysenko) [inline client/a8c-for-agencies/sections/partner-directory/agency-details/hooks/use-country-list.ts:1]: Could we possibly re-use `client/a8c-for-agencies/sections/signup/agency-details-form/hooks/use-countries-and-states.ts` ?

Comment (cleacos) [reply client/a8c-for-agencies/sections/partner-directory/agency-details/hooks/use-country-list.ts:1]: I copied and pasted it, but removed the state part. Really I think we could use it, even with the state part.

Comment (cleacos) [reply client/a8c-for-agencies/sections/partner-directory/agency-details/hooks/use-country-list.ts:1]: and the Stripe countries check part, I was not sure if that would affect this form with fewer countries.

**Review by andrii-lysenko — APPROVED** [review]
Looks good to me codewise and works as described in testing steps 👍 

We could do some refactoring in the future, to implement unified countries hook.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/14593439

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @cleacos for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/14593439).

**Pull request merged**